tests/e2e/specs/delegate-spec.js

```diff
@@ -0,0 +1,79 @@
+			delegatePage.getTable().contains(shortenedDelegateAddress, { timeout: 30 * 1e3 });
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('not.be.checked');
+
+			// check mint
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).siblings().click();
+			delegatePage.confirmMetamaskTransaction();
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('not.be.checked');
+
+			// check all
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).siblings().click();
+			delegatePage.confirmMetamaskTransaction();
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('be.disabled');
+
+			// uncheck all
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).siblings().click();
+			delegatePage.confirmMetamaskTransaction();
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('not.be.checked');
+			cy.findByTestId(`checkbox-${delegateAddress}-all`).should('not.be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-mint`).should('not.be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-burn`).should('not.be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-claim`).should('not.be.disabled');
+			cy.findByTestId(`checkbox-${delegateAddress}-trade`).should('not.be.disabled');
```

Comment:
Shouldn't this be a separate test? something like `it('should undelegate powers from an account')`?

I would also separate tests for specific delegation methods like `burn`, `mint`, `claim`, `trade` and `all` inside tests; something like this:
```
	context('Delegate', () => {
		it(`should be able to delegate 'burn' powers`, () => {
			...
		});
		it(`should be able to delegate 'mint' powers`, () => {
			...
		});
		it(`should be able to delegate 'claim' powers`, () => {
			...
		});
		it(`should be able to delegate 'trade' powers`, () => {
			...
		});
		it(`should be able to delegate 'all' powers`, () => {
			...
		});
	})
```

**However**, the best solution would be to split this test to two different spec files like `delegation-spec.js` and `undelegation-spec.js` and do the preparation for the tests in `before()` (delegations for `undelegation-spec.js`) if needed. Tests shouldn't depend on each other and they will run in parallel to save time resulting in faster finish time than it would be in one spec file without `before()`.

1. `delegation-spec.js`
```
describe('Delegation tests', () => {
	before(() => {
		...
	});
	context('Delegate', () => {
		it(`should be able to delegate 'burn' powers`, () => {
			...
		});
		it(`should be able to delegate 'mint' powers`, () => {
			...
		});
		it(`should be able to delegate 'claim' powers`, () => {
			...
		});
		it(`should be able to delegate 'trade' powers`, () => {
			...
		});
		it(`should be able to delegate 'all' powers`, () => {
			...
		});
	});
});
```
2. `undelegation-spec.js`
```
describe('Undelegation tests', () => {
	before(() => {
		... do the preparation (delegations)
	});
	context('Undelegate', () => {
		it(`should be able to undelegate 'burn' powers`, () => {
			...
		});
		it(`should be able to undelegate 'mint' powers`, () => {
			...
		});
		it(`should be able to undelegate 'claim' powers`, () => {
			...
		});
		it(`should be able to undelegate 'trade' powers`, () => {
			...
		});
		it(`should be able to undelegate 'all' powers`, () => {
			...
		});
	});
});
```

Reply:
Gotcha. I preferred to test all the cases in 1 run to reduce the test run time / no of txns. Thoughts?

Reply:
2 tests run in parallel will be faster than 1 test run sequentially.

Reply:
Ok. I've done a bit of refactoring but it seems a conflict arises when the tests are run in parallel because am guessing since each of them does execute a transaction, I've turned that flag off.